test/html/test.ml

```diff
@@ -107,16 +107,49 @@ let () =
     let reference_file = expect_directory // lang // (file_title ^ ".html") in
 
     let validate_html file =
+      let label = "tidy" in
+      let muted_warnings = String.concat "," [
+        "NESTED_EMPHASIS";
+        "MISSING_STARTTAG";
+        "DISCARDING_UNEXPECTED";
+      ] in
       let tidy = String.concat " " [
         "tidy";
         "-quiet";
+        "--mute " ^ muted_warnings;
         "--mute-id yes";
-        "--show-errors 20";
+        "--show-errors 200";
         "-errors";
         "-ashtml"
       ]
       in
-      command tidy "%s %s" tidy file;
+      let cmd = Printf.sprintf "%s %s" tidy file in
+      let (stdout, stdin, stderr) = Unix.open_process_full cmd [||]  in
+      let errors =
+          let rec r acc = match input_line stderr with
+          | s -> r ((s ^ "\n") :: acc)
+          | exception End_of_file -> List.rev acc
+          in
+          r []
+      in
+      let exit_status = Unix.close_process_full (stdout, stdin, stderr) in
+      match exit_status with
```

Comment:
Code-style I'm not a fan of nested matches like this, but I haven't seen the codebase rely too much on operators (`>>=`, `<|>`) and I figured I wouldn't start now.

Also first time consuming outputs from a process with `Unix.*` so let me know if there's a more straightforward way of doing this!

Reply:
The `Unix.*` looks fine. For the nested matches, you could do

```ocaml
match exit_status, errors with
| WEXITED 0, _
| WEXITED 1, [] ->
  ()
| WEXITED _, _ ->
  ...
| _, _ ->
  ...
```


One of that advantages is that none of the case code will be duplicated.